lib/msal-node/src/network/HttpClient.ts

```diff
@@ -20,13 +20,16 @@ import https from "https";
 export class HttpClient implements INetworkModule {
     private proxyUrl: string;
     private customAgentOptions: http.AgentOptions | https.AgentOptions;
+    private managedIdentity: boolean;
```

Comment:
HttpClient should NOT know anything about Managed Identity. This is a violation of the Single Responsability Principle. 

At most, HTTPClient should know that it has to use the "http" module vs the "https" module. 

Can't you just look at the request URL to tell if the request URL starts with "http" and not "https" ?